Source/Core/DolphinQt2/Resources.cpp

```diff
@@ -2,46 +2,64 @@
 // Licensed under GPLv2+
 // Refer to the license.txt file included.
 
+#include <algorithm>
+
+#include <QGuiApplication>
+#include <QScreen>
 #include <QStringList>
 
 #include "Common/CommonPaths.h"
 #include "Common/FileUtil.h"
 #include "DolphinQt2/Resources.h"
+#include "DolphinQt2/Settings.h"
 
 QList<QPixmap> Resources::m_platforms;
 QList<QPixmap> Resources::m_countries;
 QList<QPixmap> Resources::m_ratings;
 QList<QPixmap> Resources::m_misc;
 
+QString Resources::GetPreferredPixmapScale(int max)
```

Comment:
What do you mean "worse"? And because it separates the logic of "get this icon" from "knowing which scale versions exist on the filesystem", which doesn't belong in places like the `Toolbar` class.